src/pybind/tests/test_kaldi_pybind.py

```diff
@@ -53,6 +54,21 @@ def test_matrix_reader_writer(self):
         print("Read id: {}".format(matrix_reader.Key()))
         print("Read matrix: {}".format(matrix_reader.Value()))
 
+    def test_matrix_reader_iterator(self):
+        print("=====Testing Matrix Reader Iterator=====")
+        kp_matrix = kp.FloatMatrix(2,3)
+        wspecifier = "ark,t:test.ark"
+        rspecifier = "ark:test.ark"
+        matrix_writer = kp.BaseFloatMatrixWriter_Matrix(wspecifier)
+        print("Write id: 'id_1'")
+        print("Write matrix: [0 0 0; 0 0 0]")
+        matrix_writer.Write("id_1", kp_matrix)
+        matrix_writer.Close()
+
+        for key, value in kaldi.ReaderIterator(kaldi.SequentialMatrixReader(rspecifier)):
+            print(key, value)
```

Comment:
use some sort of  self.assert*** methods to check what you read back is equal to what you have written.

`print` is for human and `assert` is for the machine.